pkg/api/norman/customization/clusterregistrationtokens/import.go

```diff
@@ -46,9 +47,40 @@ func (ch *ClusterImport) ClusterImportHandler(resp http.ResponseWriter, req *htt
 		cluster, _ = ch.Clusters.Get(clusterID, metav1.GetOptions{})
 	}
 
+	// add additional tolerations and override affinity/resource limits if user has specified. If affinity is not
+	// overridden, use default.
+	tolerations := getTolerations(cluster)
+	affinity := getAffinity(cluster)
+	resourceRequirements := getResourceRequirements(cluster)
+
 	if err = systemtemplate.SystemTemplate(resp, image.Resolve(settings.AgentImage.Get()), authImage, "", token, url,
-		false, cluster, nil, nil, nil); err != nil {
+		false, cluster, nil, tolerations, affinity, resourceRequirements, nil); err != nil {
 		resp.WriteHeader(500)
 		resp.Write([]byte(err.Error()))
 	}
 }
+
+func getTolerations(cluster *apimgmtv3.Cluster) []corev1.Toleration {
```

Comment:
Jake already covered the duplicate functions, my two cents is to remove it from this package and instead place it in a more common package since we will need them throughout Rancher. private registries have similar functions and they're placed in a common `pkg/cluster/private_registries.go` file.